accounts/views.py

```diff
@@ -77,6 +77,18 @@ def crash_me(request):
     raise Exception
 
 
+def check_username(request):
+    username = request.GET.get('username', None)
+    username_valid = False
+    if username:
+        try:
+            user = User.objects.get(username__iexact=username)
+        except User.DoesNotExist:
+            username_valid = True
+    return HttpResponse(json.dumps({'result': username_valid}),
```

Comment:
Why not JsonResponse?

Btw, "readability" versus "speed" is always the worst kind of reason to choose a certain method if you don't talk about relative speed. ;) That said I'm not holy about { } versus dict( )